pmd-apex/src/main/java/net/sourceforge/pmd/lang/apex/rule/security/ApexDangerousMethodsRule.java

```diff
@@ -35,14 +50,49 @@ public Object visit(ASTUserClass node, Object data) {
             return data;
         }
 
+        collectBenignVariables(node);
```

Comment:
In that case, you are not generating False Positives, but are preventing stale Strings from being garbage collected. I'd still recommend clearing it as a final step when done visiting the `ASTUserClass`.